### PR TITLE
[geometry] Add a DOMMatrix2dInit dictionary type

### LINKED ISSUE #210: [geometry] Add a 2d-only matrix dictionary type
https://drafts.fxtf.org/geometry/#dom-dommatrix-setmatrixvalue-transformlist-transformlist

> dictionary DOMMatrixInit \{
>     unrestricted double a;
>     unrestricted double b;
>     unrestricted double c;
>     unrestricted double d;
>     unrestricted double e;
>     unrestricted double f;
>     unrestricted double m11;
>     unrestricted double m12;
>     unrestricted double m13 =​ 0;
>     unrestricted double m14 =​ 0;
>     unrestricted double m21;
>     unrestricted double m22;
>     unrestricted double m23 =​ 0;
>     unrestricted double m24 =​ 0;
>     unrestricted double m31 =​ 0;
>     unrestricted double m32 =​ 0;
>     unrestricted double m33 =​ 1;
>     unrestricted double m34 =​ 0;
>     unrestricted double m41;
>     unrestricted double m42;
>     unrestricted double m43 =​ 0;
>     unrestricted double m44 =​ 1;
>     boolean is2D;
> \};

In https://github.com/whatwg/html/issues/2826 there is a need identified to throw away 3d stuff of a matrix given as argument to `setTransform()` in 2d canvas.

I think a new dictionary type for that could be added to do this, something like

```
dictionary DOMMatrix2dInit {
    unrestricted double a;
    unrestricted double b;
    unrestricted double c;
    unrestricted double d;
    unrestricted double e;
    unrestricted double f;
    unrestricted double m11;
    unrestricted double m12;
    unrestricted double m21;
    unrestricted double m22;
    unrestricted double m41;
    unrestricted double m42;
};
```
...with "validate and fixup" defined as appropriate, and the current `DOMMatrixInit` can inherit from `DOMMatrix2dInit` I believe.

(I don't know if SVG could use this as well? cc @fsoder @AmeliaBR )

Thoughts?

### PR DESCRIPTION
**Pull request by zcorpan**
This is needed for CanvasRenderingContext2D's setTransform().

Fixes #210.

Tests: https://github.com/w3c/web-platform-tests/pull/7007

---

cc @dcrousso @junov @annevk @grorg

### CODE REVIEW

File: geometry/Overview.bs [hunk 1]
```diff
@@ -865,59 +868,61 @@ prose.
...
-    [=SameValueZero=]({{DOMMatrixInit/f}}, {{DOMMatrixInit/m42}}) is <code>false</code>.
+    <li><p>{{DOMMatrix2dInit/f}} and {{DOMMatrix2dInit/m42}} are both present and
+    [=SameValueZero=]({{DOMMatrix2dInit/f}}, {{DOMMatrix2dInit/m42}}) is <code>false</code>.
 
-    <li><p>{{DOMMatrixInit/is2D}} is <code>true</code> and at least one of {{DOMMatrixInit/m13}},
-    {{DOMMatrixInit/m14}}, {{DOMMatrixInit/m23}}, {{DOMMatrixInit/m24}}, {{DOMMatrixInit/m31}},
-    {{DOMMatrixInit/m32}}, {{DOMMatrixInit/m34}}, {{DOMMatrixInit/m43}} are present with a value
-    other than ''0'' or ''-0'', or at least one of {{DOMMatrixInit/m33}}, {{DOMMatrixInit/m44}} are
-    present with a value other than ''1''.
+    <li><p><var>dict</var> is a {{DOMMatrixInit}} and {{DOMMatrixInit/is2D}} is <code>true</code>
```
Comment (annevk) [inline geometry/Overview.bs]: There's no type checking on dictionaries.

File: geometry/Overview.bs [hunk 2]
```diff
@@ -865,59 +868,61 @@ prose.
...
+    and at least one of {{DOMMatrixInit/m13}}, {{DOMMatrixInit/m14}}, {{DOMMatrixInit/m23}},
+    {{DOMMatrixInit/m24}}, {{DOMMatrixInit/m31}}, {{DOMMatrixInit/m32}}, {{DOMMatrixInit/m34}},
+    {{DOMMatrixInit/m43}} are present with a value other than ''0'' or ''-0'', or at least one of
+    {{DOMMatrixInit/m33}}, {{DOMMatrixInit/m44}} are present with a value other than ''1''.
    </ul>
 
-  <li><p>If {{DOMMatrixInit/m11}} is not present then set it to the value of member
-  {{DOMMatrixInit/a}}, or value ''1'' if {{DOMMatrixInit/a}} is also not present.
+  <li><p>If <var>dict</var> is not a {{DOMMatrixInit}}, then return.
```
Comment (annevk) [inline geometry/Overview.bs]: Same.

File: geometry/Overview.bs (hunk 1 above)
Comment (zcorpan) [inline geometry/Overview.bs]: I tried to fix this.

**Comment by annevk** [comment]
So the idea is that you always need to pass a type? I think it would be better if you made that a boolean argument, potentially with a default value. Gives less room for confusion.

**Comment by annevk** [comment]
Looks okay to me.

**Pull request merged**